include/boost/geometry/algorithms/detail/intersection/areal_areal.hpp

```diff
@@ -80,14 +78,11 @@ struct intersection_areal_areal_<TupledOut, tupled_output_tag>
                 TupledOut
             >::type;
 
-        boost::ignore_unused
-            <
-                geometry::detail::expect_output
-                    <
-                        Areal1, Areal2, single_out,
-                        point_tag, linestring_tag, polygon_tag
-                    >
-            >();
+        (void)sizeof(geometry::detail::expect_output
+                <
+                    Areal1, Areal2, single_out,
+                    point_tag, linestring_tag, polygon_tag
+                >);
```

Comment:
This I don't like so much. `ignore_unused` was meant exactly to avoid this cryptic statement.
Is it solveable in another way?

(Why is the whole statement there anyway - but that is another question...)

Reply:
I will test taking a pointer of this type and assign it to ignore.